api/v1alpha3/azuremachine_types.go

```diff
@@ -66,6 +66,11 @@ type AzureMachineSpec struct {
 	// +optional
 	UserAssignedIdentities []UserAssignedIdentity `json:"userAssignedIdentities,omitempty"`
 
+	// RoleAssignmentName is the name of the role assignment to create for a system assigned identity. It can be any valid GUID.
```

Comment:
Did you consider allowing any string to be specified and only generating a GUID as a default?

Reply:
It has to be a valid GUID, that's an Azure requirement

Reply:
Oh... You are right. I was thrown off b/c ARM usually calls things that are names, as strings rather than UUIDs.

Azure SDK docs:
```go
// Create creates a role assignment.
// Parameters:
// scope - the scope of the role assignment to create. The scope can be any REST resource instance. For
// example, use '/subscriptions/{subscription-id}/' for a subscription,
// '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}' for a resource group, and
// '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/{resource-provider}/{resource-type}/{resource-name}'
// for a resource.
// roleAssignmentName - the name of the role assignment to create. It can be any valid GUID.
// parameters - parameters for the role assignment.
```
Ok. Carry on.